python-frontend/typeshed_serializer/runners/tox_runner.py

```diff
@@ -0,0 +1,145 @@
+    return sorted([*filenames, *resources, *config_files])
+
+
+def normalize_text_files(file_name: str) -> bytes:
+    normalized_file = Path(file_name).read_text().strip().replace('\r\n', '\n').replace('\r', '\n')
+    return bytes(normalized_file, 'utf-8')
+
+
+def read_file(file_name: str) -> bytes:
+    return Path(file_name).read_bytes()
+
+
+def compute_checksum(file_names: list[str], get_file_bytes: Callable[[str], bytes]) -> str:
+    _hash = hashlib.sha256()
+    for fn in file_names:
+        with contextlib.suppress(IsADirectoryError):
+            _hash.update(get_file_bytes(fn))
+    return _hash.hexdigest()
+
+
+def read_previous_checksum(checksum_file: str) -> Tuple[Optional[str], Optional[str]]:
+    def empty_str_to_none(s: str) -> Optional[str]:
+        if not s:
+            return None
+        return s
+
+    if not Path(checksum_file).is_file():
+        return None, None
+    with open(checksum_file, 'r') as file:
+        source_checksum = empty_str_to_none(file.readline().strip())
+        binaries_checksum = empty_str_to_none(file.readline().strip())
+        return source_checksum, binaries_checksum
+
+
+def update_checksum():
+    with open(CHECKSUM_FILE, 'w') as file:
+        source_file_names = fetch_source_file_names(SERIALIZER_PATH)
+        source_checksum = compute_checksum(source_file_names, normalize_text_files)
+        binary_file_names = fetch_binary_file_names()
+        binary_checksum = compute_checksum(binary_file_names, read_file)
+        file.writelines([f"{source_checksum}\n", binary_checksum])
+
+
+def main():
+    source_files = fetch_source_file_names(SERIALIZER_PATH)
+    (previous_sources_checksum, previous_binaries_checksum) = read_previous_checksum(CHECKSUM_FILE)
+    current_sources_checksum = compute_checksum(source_files, normalize_text_files)
+    logger.info("STARTING TYPESHED SOURCE FILE CHECKSUM COMPUTATION")
+    logger.info(f"Previous checksum {previous_sources_checksum}")
+    logger.info(f"Current checksum {current_sources_checksum}")
+    logger.info(f"Checksum is computed over {len(source_files)} files")
+    if previous_sources_checksum != current_sources_checksum:
+        logger.info("STARTING TYPESHED SERIALIZATION")
+        subprocess.run(["tox"])
+    else:
+        binary_file_names = fetch_binary_file_names()
+        current_binaries_checksum = compute_checksum(binary_file_names, read_file)
+        logger.info("STARTING TYPESHED BINARY FILES CHECKSUM COMPUTATION")
+        logger.info(f"Previous binaries checksum {previous_binaries_checksum}")
+        logger.info(f"Current binaries checksum {current_binaries_checksum}")
+        logger.info(f"Checksum is computed over {len(binary_file_names)} files")
+        if previous_binaries_checksum != current_binaries_checksum:
+            raise RuntimeError('INCONSISTENT BINARY CHECKSUMS')
+        logger.info("SKIPPING TYPESHED SERIALIZATION")
+        subprocess.run(['tox','-e', 'testenv'])
```

Comment:
As discussed, this should run `tox -e py39` instead. In MacOS `testenv` doesn't seem to be a valid environment